### PR TITLE
Cafebot nodejs sample cleanup

### PR DESCRIPTION
**Pull request by vishwacsena**
Current cafe bot sample had bugs and was incomplete. This PR addresses the gaps - 
- Re-structured code to make it easier to parse
- Updated model files 
- Updated msbot clone recipe
- Updated README to clarify instructions to get set up. 
- Removed bunch of dead code

### CODE REVIEW

File: samples/javascript_nodejs/51.cafe-bot/dialogs/whatCanYouDo/resources/whatCanYouDoCard.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
             "choices": [
                 {
                     "title": "book a table for 2 in seattle",
-                    "value": " {\"intent\": \"Book_Table\", \"cafeLocation\": \"seattle\", \"number\": \"2\", \"text\": \"book a table for 2 in seattle\"}"
```
Comment (daveta) [inline samples/javascript_nodejs/51.cafe-bot/dialogs/whatCanYouDo/resources/whatCanYouDoCard.json]: >cafeLocation [](start = 62, length = 12)

Curious why this was removed?

File: samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/interruptionDispatcher.js [hunk 2]
```diff
@@ -37,23 +42,33 @@ module.exports = {
             // add dialogs
             this.addDialog(new WhatCanYouDoDialog());
             this.addDialog(new QnADialog(botConfig, userProfileAccessor));
+
+            // add recognizer
+            const luisConfig = botConfig.findServiceByNameOrId(LUIS_CONFIGURATION);
```
Comment (daveta) [inline samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/interruptionDispatcher.js]: >LUIS_CONFIGURATION [](start = 63, length = 18)

I'm not sure how you are dealing with the msbot decoration that occurs on LUIS services in the generated .bot configuration file.
For example, if your msbot name that was deployed is called "cafe", your corresponding LUIS name in the generated .bot configuration should be cafe_<my_luis_name>.

File: samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/mainDispatcher.js [hunk 3]
```diff
@@ -57,6 +61,16 @@ module.exports = {
             this.addDialog(new QnADialog(botConfig, this.userProfileAccessor));
             this.addDialog(new WhoAreYouDialog(botConfig, conversationState, this.userProfileAccessor, onTurnAccessor, this.reservationAccessor));
             this.addDialog(new BookTableDialog(botConfig, this.reservationAccessor, onTurnAccessor, this.userProfileAccessor, conversationState));
+
+            // add recognizer
+            const luisConfig = botConfig.findServiceByNameOrId(LUIS_CONFIGURATION);
+            if (!luisConfig || !luisConfig.appId) throw new Error(`Cafe Dispatch LUIS model not found in .bot file. Please ensure you have all required LUIS models created and available in the .bot file. See readme.md for additional information.\n`);
+            this.luisRecognizer = new LuisRecognizer({
```
Comment (daveta) [inline samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/mainDispatcher.js]: >LuisRecognizer [](start = 38, length = 14)

In C# side, we housed the recognizers into singleton objects to (in theory) avoid recreation costs on every turn but more to (in theory) maintain the connection..

File: samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/mainDispatcher.js [hunk 4]
```diff
@@ -85,32 +99,44 @@ module.exports = {
          *
          * This method examines the incoming turn property to determine
          * 1. If the requested operation is permissible - e.g. if user is in middle of a dialog,
-         *     then an out of order reply should not be allowed.
+         *    then an out of order reply should not be allowed.
```
Comment (daveta) [inline samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/mainDispatcher.js]: > then an out of order reply should not be allowed. [](start = 13, length = 50)

I'm trying to square this statement with interruptions.  Is that what this is suggesting?

File: samples/javascript_nodejs/51.cafe-bot/dialogs/whatCanYouDo/resources/whatCanYouDoCard.json (hunk 1 above)
Comment (vishwacsena) [reply samples/javascript_nodejs/51.cafe-bot/dialogs/whatCanYouDo/resources/whatCanYouDoCard.json]: as a sample, I just wanted the dialog to start with asking for location. No other significant reason.

File: samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/mainDispatcher.js (hunk 4 above)
Comment (vishwacsena) [reply samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/mainDispatcher.js]: That's right.

File: samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/interruptionDispatcher.js (hunk 2 above)
Comment (vishwacsena) [reply samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/interruptionDispatcher.js]: I updated the readme to promote use of the CLI tools to get setup with the services and if you do, the names here (line 17) will match the instructions. Of course if you choose a different name (or need to pick up the LUIS app name from msbot), you will see an error on launch and need to update line 17.

**Pull request merged**